src/main/java/com/exasol/adapter/dynamodb/DynamodbAdapter.java

```diff
@@ -112,24 +108,41 @@ public GetCapabilitiesResponse getCapabilities(final ExaMetadata exaMetadata,
 	@Override
 	public PushDownResponse pushdown(final ExaMetadata exaMetadata, final PushDownRequest request)
```

Comment:
Method to complex. Use method extraction to cut it down to smaller pieces on the same abstraction level.